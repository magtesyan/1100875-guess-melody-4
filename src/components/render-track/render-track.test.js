import React from "react";
import RenderTrack from "./render-track.jsx";
import renderer from "react-test-renderer";

const userAnswers = [false, true, false, false];
const answer = {
  id: `12345`,
  src: `https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg`,
  genre: `blues`,
};
const index = 1;

it(`ArtistQuestionScreen is rendered correctly`, () => {
  const handleAnswerSubmit = jest.fn();
  const tree = renderer
    .create(<RenderTrack
      src = {answer.src}
      index = {index}
      key = {answer.id}
      userAnswers = {userAnswers}
      handleAnswerSubmit = {handleAnswerSubmit}
    />
    ).toJSON();

  expect(tree).toMatchSnapshot();
});
