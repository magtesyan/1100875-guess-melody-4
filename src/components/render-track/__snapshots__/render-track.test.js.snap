// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen is rendered correctly 1`] = `
<div
  className="track"
>
  <button
    className="track__button track__button--play"
    type="button"
  />
  <div
    className="track__status"
  >
    <audio
      src="https://upload.wikimedia.org/wikipedia/commons/4/4e/BWV_543-fugue.ogg"
    />
  </div>
  <div
    className="game__answer"
  >
    <input
      checked={true}
      className="game__input visually-hidden"
      id="answer-1"
      name="answer"
      onChange={[Function]}
      type="checkbox"
      value="answer-1"
    />
    <label
      className="game__check"
      htmlFor="answer-1"
    >
      Отметить
    </label>
  </div>
</div>
`;
